testkit/src/main/scala/akka/kafka/testkit/javadsl/KafkaTest.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 - 2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.kafka.testkit.javadsl
+
+import java.util.concurrent.{CompletionStage, TimeUnit}
+
+import akka.Done
+import akka.japi.Pair
+import akka.kafka.Subscriptions
+import akka.kafka.javadsl.Consumer
+import akka.kafka.scaladsl.Producer
+import akka.kafka.testkit.internal.KafkaTestKit
+import akka.stream.Materializer
+import akka.stream.javadsl.{Keep, Sink}
+import akka.stream.scaladsl.Source
+import akka.stream.testkit.javadsl.StreamTestKit
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.junit.jupiter.api._
+import org.slf4j.{Logger, LoggerFactory}
+
+import scala.compat.java8.FutureConverters._
+
+/**
+ * JUnit 5 aka Jupiter base-class with some convenience for accessing
+ * a Kafka broker.
+ * Extending classes must be annotated with `@TestInstance(Lifecycle.PER_CLASS)` to create
+ * a single instance of the test class with `@BeforeAll` and `@AfterAll` annotated methods called
+ * by the test framework.
+ */
+abstract class KafkaTest extends KafkaTestKit {
+
+  val log: Logger = LoggerFactory.getLogger(getClass)
+
+  final val partition0 = 0
+
+  def materializer: Materializer
+
+  /**
+   * Sets up the Admin client. Override if you want custom initialization of the admin client.
+   */
+  @BeforeAll def setUpAdmin() = setUpAdminClient()
+
+  /**
+   * Cleans up the Admin client. Override if you want custom cleaning up of the admin client.
+   */
+  @AfterAll def cleanUpAdmin() = cleanUpAdminClient()
+
+  @AfterEach def checkForStageLeaks() = StreamTestKit.assertAllStagesStopped(materializer)
+
+  /**
+   * Overwrite to set different default timeout for [[KafkaTest.resultOf]].
+   */
+  def resultOfTimeout: java.time.Duration = java.time.Duration.ofSeconds(5)
+
+  def produceString(topic: String, messageCount: Int, partition: Int): CompletionStage[Done] =
```

Comment:
Common code between JUnit4 and Junit5 tests should be moved to a common abstract class.